airbyte-integrations/connectors/source-pipedrive/source_pipedrive/streams.py

```diff
@@ -0,0 +1,173 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union
+
+import pendulum
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+PIPEDRIVE_URL_BASE = "https://api.pipedrive.com/v1/"
+
+
+class PipedriveStream(HttpStream, ABC):
+    url_base = PIPEDRIVE_URL_BASE
+    primary_key = "id"
+    data_field = "data"
+    page_size = 50
+
+    def __init__(self, api_token: str, replication_start_date: pendulum.datetime = None, **kwargs):
+        super().__init__(**kwargs)
+        self._api_token = api_token
+        self._replication_start_date = replication_start_date
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        if self._replication_start_date:
```

Comment:
if everything can be incremental why not just do it by default from 1 year ago? e.g: `self._replication_start_date = replication_start_date or pendulum.now().minus(years=1)`? it's always better to do incremental sync when possible

Reply:
Sure, we can do it. I'll make an update :)

Reply:
@sherifnada this default should be in the spec, not in the code